src/blocks/opentable/edit.js

```diff
@@ -0,0 +1,204 @@
+/* global coblocksBlockData */
+
+/**
+ * External dependencies
+ */
+import { OpentableIcon as icon } from '@godaddy-wordpress/coblocks-icons';
+import { debounce } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Controls from './controls';
+import OpenTable from './opentable';
+import InspectorControls from './inspector';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect, useCallback } from '@wordpress/element';
+import { compose, usePrevious } from '@wordpress/compose';
+import {
+	Placeholder,
+	Spinner,
+	Notice,
+	Button,
+	withNotices,
+	Icon,
+	FormTokenField,
+} from '@wordpress/components';
+
+const Edit = ( props ) => {
+	const [ ridField, setRidField ] = useState( [] );
+	const [ queryResults, setQueryResults ] = useState( [] );
+	const { className, attributes, noticeUI, noticeOperations } = props;
+	const [ noResultsFound, setNoResultsFound ] = useState( false );
+	const [ isEditing, setIsEditing ] = useState( ! attributes.restaurantIDs.length );
+	const [ isLoading, setIsLoading ] = useState( false );
+
+	const prevIDs = usePrevious( attributes.restaurantIDs );
+
+	const getLocaleCode = () => {
+		if ( typeof coblocksBlockData.localeCode !== 'undefined' ) {
+			return coblocksBlockData.localeCode;
+		}
+		return '';
+	};
+
+	//searches the opentable reservation network for restaurants with the given name or ID
+	const searchRestaurants = useCallback( debounce( ( token ) => {
+		setIsLoading( true );
+		noticeOperations.removeAllNotices();
+		fetch(
+			'https://www.opentable.com/widget/1reservation/restaurant-search?pageSize=15' +
```

Comment:
Is `widget/1reservation` a typo? I think this should be `widget/reservation`.

Reply:
Oops. That shouldn’t have made it into a commit - that’s how I was testing the error handling.